hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/metrics/RpcMetrics.java

```diff
@@ -49,7 +51,7 @@
   final String name;
   final boolean rpcQuantileEnable;
   /** The time unit used when storing/accessing time durations. */
-  public final static TimeUnit TIMEUNIT = TimeUnit.MILLISECONDS;
+  private static TimeUnit metricsTimeUnit = TimeUnit.MILLISECONDS;
```

Comment:
I think the variable can be final.

Reply:
The problem is we are reassigning it at L113 so it could not be made final

Reply:
I also thought of using AtomicReference to make this final but then realized this is not frequently updated, hence using AtomicReference would be of no use as we are not much worried about atomically retrieving the value.

Reply:
Can we set the timeunit in the constructor?

Reply:
Since this is static, setting in constructor might not be that accurate (also it produces spotbugs, as per first QA result).

Reply:
Let me do one thing here. I can ensure this metric is updated only once for first RPC.Server initialization.

Reply:
Is the time unit must be static? There are a few RPC servers in a JVM in most cases, so the performance impact seems low.

Reply:
I think the code will be simple if the time unit is non-static and final.

Reply:
I agree that it would be simpler but some of the consumers e.g `DecayRpcScheduler`, `RpcScheduler` don't have access to `Server` or `RpcMetrics` and hence they are able to use metrics timeUnit as static reference. I tried to see if DecayRpcScheduler can access RpcMetrics object but it seems difficult because they are independent entities (I might be missing something though). Any suggestion?

On the other hand, I think having timeUnit static can be advantageous because all RpcMetrics can follow same timeunit based on core-site config `rpc.metrics.timeunit`.

Reply:
I just realized that default implementation of `RpcScheduler` that requires metric unit is just to maintain backward compatibility with old implementations hence we can let it use default timeunit. Since all new impl are supposed to implement `addResponseTime()`, I think we can make this non-static in RpcMetrics.
Let me make this change.

Reply:
@aajisaka I have addressed this in latest revision.

Reply:
Thanks!